baybe/campaign.py

```diff
@@ -222,6 +265,72 @@ def add_measurements(
             numerical_measurements_must_be_within_tolerance,
         )
 
+    def _mark_as_measured(
+        self,
+        measurements: pd.DataFrame,
+        numerical_measurements_must_be_within_tolerance: bool,
+    ) -> None:
+        """Mark the given elements of the discrete subspace as measured.
+
+        Args:
+            measurements: A dataframe containing parameter configurations to be
+                marked as measured.
+            numerical_measurements_must_be_within_tolerance: See
+                :func:`baybe.utils.dataframe.fuzzy_row_match`.
+        """
+        idxs_matched = fuzzy_row_match(
+            self.searchspace.discrete.exp_rep,
+            measurements,
+            self.parameters,
+            numerical_measurements_must_be_within_tolerance,
+        )
+        self._searchspace_metadata.loc[idxs_matched, _MEASURED] = True
+
+    def toggle_discrete_candidates(
+        self,
+        filter: pd.DataFrame,
+        exclude: bool,
+        anti: bool = False,
+        dry_run: bool = False,
```

Comment:
Why did you include this? There is no point in the code where you have `dry_run=True` (according to VS Code Search), so is this just for people to play around with the function?

Reply:
Correct, there is a small chance that this will ever be used in production code (might come handy for testing, though). But the main reason is like you said and what is also the purpose of other "dry_run" flags (like `git clean -n`): to play around without messing up something. It can be painful otherwise to figure out your correct filtering mechanism since, when passing an incorrect one, it's already too late: your previous state is lost irreversibly.

But if both of you say that it's not needed, I'm happy to remove it. Thus, let's do the regular voting (@Scienfitz). Thumbs up for keep, thumbs down for remove.

Reply:
Nope, fine for me. Was just wanting to make sure that I did not overlook something.